rcl_yaml_param_parser/src/parser.c

```diff
@@ -744,17 +744,17 @@ static void * get_value(
     return ret_val;
   }
 
-  if ((0 == strncmp(value, "N", strlen(value))) ||
-    (0 == strncmp(value, "n", strlen(value))) ||
-    (0 == strncmp(value, "no", strlen(value))) ||
-    (0 == strncmp(value, "No", strlen(value))) ||
-    (0 == strncmp(value, "NO", strlen(value))) ||
-    (0 == strncmp(value, "false", strlen(value))) ||
-    (0 == strncmp(value, "False", strlen(value))) ||
-    (0 == strncmp(value, "FALSE", strlen(value))) ||
-    (0 == strncmp(value, "off", strlen(value))) ||
-    (0 == strncmp(value, "Off", strlen(value))) ||
-    (0 == strncmp(value, "OFF", strlen(value))))
+  if ((0 == strcmp(value, "N")) ||
+    (0 == strcmp(value, "n")) ||
+    (0 == strcmp(value, "no")) ||
+    (0 == strcmp(value, "No")) ||
+    (0 == strcmp(value, "NO")) ||
+    (0 == strcmp(value, "false")) ||
+    (0 == strcmp(value, "False")) ||
+    (0 == strcmp(value, "FALSE")) ||
+    (0 == strcmp(value, "off")) ||
+    (0 == strcmp(value, "Off")) ||
+    (0 == strcmp(value, "OFF")))
```

Comment:
On an unrelated note, why do we need to determine this for ourselves? Shouldn't libyaml tell us the YAML type of something given a string?

Reply:
I asked @anup-pem and apparently libyaml doesn't provide this functionality :| So never mind.

Reply:
The only problem that I'm still seeing, is the following one:
```yaml
a: 'true'
b: true
```
The first one should be recognized as an string, and the second one as a bool. Now, both are being recognized like a bool.

Reply:
I have solved the problem I described above in my last commit.